python/sdist/amici/plotting.py

```diff
@@ -64,38 +78,54 @@ def plot_observable_trajectories(
     ax: Optional[Axes] = None,
     model: Model = None,
     prefer_names: bool = True,
+    marker=None,
 ) -> None:
     """
-    Plot observable trajectories
+    Plot observable trajectories.
 
     :param rdata:
         AMICI simulation results as returned by
-        :func:`amici.amici.runAmiciSimulation`
-
+        :func:`amici.amici.runAmiciSimulation`.
     :param observable_indices:
-        Indices of observables for which trajectories are to be plotted
-
+        Indices of observables for which trajectories are to be plotted.
     :param ax:
-        matplotlib Axes instance to plot into
-
+        :class:`matplotlib.pyplot.Axes` instance to plot into.
     :param model:
-        amici model instance
-
+        The model *rdata* was generated from.
     :param prefer_names:
-        Whether observables names should be preferred over IDs, if available.
+        Whether observable names should be preferred over IDs, if available.
+    :param marker:
+        Point marker for plotting (see
+        `matplotlib documentation <https://matplotlib.org/stable/api/markers_api.html>`_).
+
     """
     if not ax:
         fig, ax = plt.subplots()
     if not observable_indices:
         observable_indices = range(rdata["y"].shape[1])
-    for iy in observable_indices:
-        if model is None:
-            label = f"$y_{{{iy}}}$"
-        elif prefer_names and model.getObservableNames()[iy]:
-            label = model.getObservableNames()[iy]
-        else:
-            label = model.getObservableIds()[iy]
-        ax.plot(rdata["t"], rdata["y"][:, iy], label=label)
+
+    if marker is None:
+        # Show marker if only one time point is available,
+        #  otherwise nothing will be shown
+        marker = "o" if len(rdata.t) == 1 else None
+
+    if model is None and rdata.ptr.observable_ids is None:
+        labels = [f"$y_{{{iy}}}$" for iy in observable_indices]
+    elif model is not None and prefer_names:
+        labels = np.asarray(model.getObservableNames())[
```

Comment:
`unindexed_observables = model.getObservableNames() if prefer_names else model.getObservableIds()` to merge `elif model is not None and prefer_names` and `elif model is not None` cases?

Reply:
I would rather keep it separate as it's already sufficiently confusing what's to be used :sweat_smile:.
We'd still need some distinction between the two cases to fall back to IDs if the name is ''.